public/app/plugins/datasource/tempo/SearchTraceQLEditor/utils.ts

```diff
@@ -81,9 +81,16 @@ const adHocValueHelper = (f: AdHocVariableFilter) => {
   if (intrinsics.find((t) => t === f.key)) {
     return f.value;
   }
+  if (parseInt(f.value, 10).toString() === f.value) {
+    return f.value;
```

Comment:
It'd be nice to use the type of the tag value, e.g.

```json
{
    "tagValues": [
        {
            "type": "int", // <---- this
            "value": "400"
        }
...
}

This could allow parsing floats and avoid parsing numbers that are stored as strings (if that even happens though 🤔 )

Reply:
Yes I think we should be this is more of a wider issue in the app right now at least and we have an issue for it [here](https://github.com/grafana/explore-traces/issues/33). Would like to a take a good look into this.

In this case I just noticed by chance that status.code was improperly typed so put in a temp fix.